src/main/java/tan/Storage.java

```diff
@@ -0,0 +1,327 @@
+package tan;
+
+
+import tan.exceptions.TaskToStringException;
+import tan.tasktype.Deadline;
+import tan.tasktype.Event;
+import tan.tasktype.Task;
+import tan.tasktype.ToDo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+
+
+/**
+ * Key Notes:
+ * Call setWriterAndReader to set the csvReader & csvWriter.
+ * Then use respective reader/writer to perform task.
+ * If you close csvWriter, you have to set the writer again
+ * and write from top of file again.
+ * Use Flush to push to current writing in buffer to file.
+ */
+public class Storage {
+    protected static BufferedReader csvReader;
+    protected static BufferedWriter csvWriter;
+    private static final String homePath = System.getProperty("user.dir");
+    private static final String[] TITLE = {"Type", "Status", "Description", "Date"};
+    private static final String FILE_NAME = "taskData.csv";
+    private static Path DATA_PATH;
+    private static int totalLinesLoaded = 0;
+
+    /**
+     * Returns a Task based on the parameters passed in. Null otherwise.
+     * This function creates & returns the actual task based on the
+     * parameters passed in. If the task type integer is not recognized,
+     * null is returned.
+     *
+     * @param taskType The integer representing its type of task.
+     * @param isDone   The completion status of the task.
+     * @param name     The Name of the task.
+     * @param date     The date, if there is, of the task.
+     * @return The task created. Null if task type is not recognized.
+     */
+    private static Task createTask(int taskType, boolean isDone, String name, LocalDate date) {
+        Task newTask;
+        switch (taskType) {
+        case 0:
+            //Task type todo.
+            newTask = new ToDo(name, isDone);
+            break;
+        case 1:
+            //Task type Deadline.
+            newTask = new Deadline(name, isDone, date);
+            break;
+        case 2:
+            //Task type event.
+            newTask = new Event(name, isDone, date);
+            break;
```

Comment:
Thank you for the advice! I initially thought of using enums to do this, but thought it might pose an issue when trying to save/load from the data file. Never though of just using the string directly. I have since changed it, Thank you!!!!